src/App.tsx

```diff
@@ -1,24 +1,217 @@
+    if (!todoTitle.trim()) {
+      setIsError(true);
+      setErrorText(ErrorNotifications.TITLE);
+    } else {
+      const newTodoBody: NewTodo = {
+        title: todoText,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      try {
+        setTodoTitle('');
+        await addTodo(USER_ID, newTodoBody);
+        getAllTodos();
+      } catch {
+        setIsError(true);
+        setErrorText(ErrorNotifications.ADD);
+      }
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setEditingTodosId(cur => [...cur, todoId]);
+    try {
+      await deleteTodo(USER_ID, todoId);
+      getAllTodos();
+    } catch {
+      setIsError(true);
+      setErrorText(ErrorNotifications.DELETE);
+    }
+
+    setEditingTodosId(cur => cur.filter(id => id !== todoId));
+  };
+
+  const handleCompletedTodo = async (todoId: number, toggle: boolean) => {
+    const completed = !toggle;
+
+    setEditingTodosId(cur => [...cur, todoId]);
+    try {
+      await completedTodo(USER_ID, todoId, { completed });
+      getAllTodos();
+    } catch {
+      setIsError(true);
+      setErrorText(ErrorNotifications.UPDATE_STATUS);
+    }
+
+    setEditingTodosId(cur => cur.filter(id => id !== todoId));
+  };
+
+  const handleDeleteCompleted = () => {
+    const completedTodosId = todos
+      .filter(todo => todo.completed === true).map(todo => todo.id);
+
+    completedTodosId.forEach(todoId => handleDeleteTodo(todoId));
+  };
+
+  const handleUpdateTodoTitle = async (
+    todoId: number,
+    newTitle: string,
+  ) => {
+    setEditingTodosId(cur => [...cur, todoId]);
+    try {
+      setDisableInput(true);
+      await updateTodoTitle(USER_ID, todoId, { title: newTitle });
+      getAllTodos();
+    } catch {
+      setIsError(true);
+      setErrorText(ErrorNotifications.UPDATE);
+    }
+
+    setEditingTodosId([]);
+  };
+
+  const toggleAll = async () => {
+    todos.forEach(todo => handleCompletedTodo(todo.id, isToggled));
+    isToggled = !isToggled;
+  };
+
+  const hasCompletedTodos = todos.some(todo => todo.completed === true);
+
+  const activeTodosCounter = todos
+    .filter(todo => todo.completed === false).length;
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const todoTitleChange = (value: string) => {
+    setTodoTitle(value);
+  };
+
+  const closeError = () => {
+    setIsError(false);
+  };
+
+  const showForm = (todoId: number) => {
+    setFormShowedForId(todoId);
+  };
+
+  const changeFilter = (selectedFilter: FilterBy) => {
+    setFilterBy(selectedFilter);
+  };
+
+  const hasActiveTodos = todos.some(todo => !todo.completed);
+
+  let filteredTodos = [...todos];
+
+  switch (filterBy) {
+    case FilterBy.Active:
+      filteredTodos = todos.filter(todo => todo.completed === false);
+      break;
+    case FilterBy.Complited:
+      filteredTodos = todos.filter(todo => todo.completed === true);
+      break;
+    case FilterBy.All:
+    default:
+      filteredTodos = todos;
+  }
```

Comment:
move functionality for getting prepared filtered todos to separate function outside of the component